src/main/java/org/opensearch/ad/task/ADTaskManager.java

```diff
@@ -560,21 +587,23 @@ public void stopDetector(
      * @param adTaskTypes AD task types
      * @param function consumer function
      * @param transportService transport service
+     * @param resetTaskState reset task state or not
      * @param listener action listener
      * @param <T> action listener response type
      */
-    public <T> void getLatestADTask(
+    public <T> void getAndExecuteOnLatestADTask(
```

Comment:
maybe rename to getAndExecuteOnLatestDetectorLevelTask for readability since you set the entity value to null?